### PR TITLE
GH-1641: As a developer I want to have a LSP command to clean the current index and output files

### PR DESCRIPTION
**Pull request by mmews-n4**
see #1641

### CODE REVIEW

File: lsp-clients/n4js-vscode-extension/src/extensionProvider.n4js [hunk 1]
```diff
@@ -55,11 +55,19 @@ export function getActivate(vscode: any+, vscodeLC: any+): any {
...
         lc.trace = VSCodeJRCP.Trace.Verbose;
-        let disposable = lc.start();
+        let disposableLangClient = lc.start();
         
         // Push the disposable to the context's subscriptions so that the 
         // client can be deactivated on extension deactivation
-        context.subscriptions.push(disposable);
+        context.subscriptions.push(disposableLangClient);
+
+        //// Example for command that is handled in the client extension ////
```
Comment (szarnekow) [inline lsp-clients/n4js-vscode-extension/src/extensionProvider.n4js]: Do we need this commented code here?

File: plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/contentassist/N4JSCandidateFilter.java [hunk 2]
```diff
@@ -10,7 +10,6 @@
  */
 package org.eclipse.n4js.ui.contentassist;
 
-import org.eclipse.n4js.scoping.utils.AbstractDescriptionWithError;
```
Comment (szarnekow) [inline plugins/org.eclipse.n4js.ui/src/org/eclipse/n4js/ui/contentassist/N4JSCandidateFilter.java]: :+1: Attention to detail is appreciated

File: tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/lsp/CommandRebuildTest.java [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+		languageServer.initialize(initParams);
+		languageServer.initialized(null);
+		languageServer.joinInitBuildFinished();
+
+		// check pre-state
+		assertEquals(0, client.getErrorsCount());
+		setFileCreationDate(prjStatePath);
+		setFileCreationDate(genFileStatePath);
+
+		// wait two seconds because file time does not consider milliseconds
```
Comment (szarnekow) [inline tests/org.eclipse.n4js.ide.tests/src/org/eclipse/n4js/ide/tests/lsp/CommandRebuildTest.java]: Obsolete comment

File: plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/commands/N4JSCommandService.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	public List<String> initialize() {
+		return Lists.newArrayList(N4JS_REBUIlD);
+	}
+
+	@Override
+	public Object execute(ExecuteCommandParams params, ILanguageServerAccess access, CancelIndicator cancelIndicator) {
+		String command = params.getCommand();
+		switch (command) {
+		case N4JS_REBUIlD:
+			lspServer.clean();
```
Comment (szarnekow) [inline plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/server/commands/N4JSCommandService.java]: I'm afraid that we'll cause a race condition by submitting to the server from a worker thread.

**Comment by mmews-n4** [comment]
Thanks!

**Pull request merged**